tests/unit/asio_service_stream_test.cxx

```diff
@@ -29,57 +29,178 @@ using namespace raft_functional_common;
 namespace asio_service_stream_test {
     const std::string TEST_MSG = "stream-test-msg-str";
 
+    class stream_statistic {
+    public:
+        stream_statistic() 
+        : resp_log_index_(0)
+        , msg_mismatch_(false)
+        , reqs_out_of_order_(false)
+        , next_log_index_(1)
+        {
+            std::string server_log_file_name = "./srv" + std::to_string(1) + ".log";
+            server_logger_ = cs_new<logger_wrapper>(server_log_file_name);
+            std::string client_log_file_name = "./srv" + std::to_string(2) + ".log";
+            client_logger_ = cs_new<logger_wrapper>(client_log_file_name);
+        }
+
+        bool waiting_for_responses(int timeout_ms = 3000) {
```

Comment:
I just realized that this needs to be renamed. Can you rename it to `wait_for_responses`? Also it will be better to modify it as follows:
```c++
int wait_for_responses(int timeout_ms = 3000) {
    TestSuite::_msg("wait for responses (up to %d ms)\n", timeout_ms);
    ea_.wait_ms(timeout_ms);
    SimpleLogger* ll = client_logger_->getLogger();
    _log_info(ll, "resp: %ld, message sent: %ld", next_log_index_.load(), num_messages_sent_);
    CHK_EQ(next_log_index_.load(), num_messages_sent_ + 1);
    return 0;
}
```
and do
```c++
CHK_Z(wait_for_responses());
```